### PR TITLE
enable ssl

### PR DESCRIPTION
**Pull request by jcantrill**
This PR enables use of SSL 

I expect this can be cherry-picked forward

cc @richm

### CODE REVIEW

**Comment by richm** [comment]
lgtm

File: lib/fluent/plugin/in_prometheus.rb [hunk 1]
```diff
@@ -23,13 +40,39 @@ def configure(conf)
...
+        }
+        if @ssl['certificate_path']
+          cert = OpenSSL::X509::Certificate.new(File.read(@ssl['certificate_path']))
+          ssl_config[:SSLCertificate] = cert
+        end
+        if @ssl['private_key_path']
+          key = OpenSSL::PKey::RSA.new(File.read(@ssl['private_key_path']))
+          ssl_config[:SSLPrivateKey] = key
+        end
+        ssl_config[:SSLCACertificateFile] if @ssl['ca_path']
```
Comment (kazegusuri) [inline lib/fluent/plugin/in_prometheus.rb]: Is this line valid?

File: lib/fluent/plugin/in_prometheus.rb [hunk 2]
```diff
@@ -23,13 +40,39 @@ def configure(conf)
...
-      )
+      }
+      unless @ssl.nil? || !@ssl['enable']
+        require 'webrick/https'
+        require 'openssl'
+        if (@ssl['certificate_path'] && @ssl['private_key_path'].nil?) || (@ssl['certificate_path'].nil? && @ssl['private_key_path'])
+            raise RuntimeError.new("certificate_path and private_key_path most both be defined")
+        end
+        ssl_config = {
+           :SSLEnable => true
```
Comment (kazegusuri) [inline lib/fluent/plugin/in_prometheus.rb]: prefer `SSLEnable: true`

File: lib/fluent/plugin/in_prometheus.rb [hunk 3]
```diff
@@ -23,13 +40,39 @@ def configure(conf)
...
-      @server = WEBrick::HTTPServer.new(
+      config = {
         BindAddress: @bind,
         Port: @port,
         MaxClients: 5,
         Logger: WEBrick::Log.new(STDERR, WEBrick::Log::FATAL),
         AccessLog: [],
-      )
+      }
+      unless @ssl.nil? || !@ssl['enable']
```
Comment (kazegusuri) [inline lib/fluent/plugin/in_prometheus.rb]: use `if` instead. `if @ssl && @ssl['enable']` ?

Comment (jcantrill) [reply lib/fluent/plugin/in_prometheus.rb]: I was unable to achieve what I expected using the 'if' form.  There always seemed to be a need to test against `nil?`

File: lib/fluent/plugin/in_prometheus.rb (hunk 1 above)
Comment (jcantrill) [reply lib/fluent/plugin/in_prometheus.rb]: No.  good catch

File: lib/fluent/plugin/in_prometheus.rb (hunk 3 above)
Comment (kazegusuri) [reply lib/fluent/plugin/in_prometheus.rb]: Ok!

**Comment by kazegusuri** [comment]
This is merged into v0.12. Please also send a PR if you need to use this feature in v1 fluentd.

**Comment by richm** [comment]
@jcantrill were you planning to submit a cherry-pick PR for the fluentd v1 branch?

**Comment by jcantrill** [comment]
@richm I did... #58
